src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -248,20 +248,56 @@ public Optional<ShortCircuitAnalysisResultEntity> findResultsWithLimitViolations
     }
 
     @Transactional(readOnly = true)
-    public Page<FaultResultEntity> findFaultResultsPage(ShortCircuitAnalysisResultEntity result, List<ResourceFilter> resourceFilters, Pageable pageable, FaultResultsMode mode) {
+    public Page<FaultResultEntity> findFaultResultsPage(ShortCircuitAnalysisResultEntity result,
+                                                        List<ResourceFilter> resourceFilters,
+                                                        Pageable pageable,
+                                                        FaultResultsMode mode) {
         Objects.requireNonNull(result);
+
+        Sort.Order secondarySort = extractSecondarySort(pageable);
+
         Specification<FaultResultEntity> specification = FaultResultSpecificationBuilder.buildSpecification(result.getResultUuid(), resourceFilters);
         // WARN org.hibernate.hql.internal.ast.QueryTranslatorImpl -
         // HHH000104: firstResult/maxResults specified with collection fetch; applying in memory!
         // cf. https://vladmihalcea.com/fix-hibernate-hhh000104-entity-fetch-pagination-warning-message/
         // We must separate in two requests, one with pagination the other one with Join Fetch
-        Page<FaultResultEntity> faultResultsPage = faultResultRepository.findAll(specification, addDefaultSort(pageable, DEFAULT_FAULT_RESULT_SORT_COLUMN));
+        Page<FaultResultEntity> faultResultsPage = faultResultRepository.findAll(
+                specification, addDefaultSort(
+                        filterOutChildrenSort(pageable, secondarySort),
+                        DEFAULT_FAULT_RESULT_SORT_COLUMN
+                ));
         if (faultResultsPage.hasContent() && mode != FaultResultsMode.BASIC) {
-            appendLimitViolationsAndFeederResults(faultResultsPage);
+            appendLimitViolationsAndFeederResults(faultResultsPage, secondarySort, resourceFilters);
         }
         return faultResultsPage;
     }
 
+    private Pageable filterOutChildrenSort(Pageable pageable, Sort.Order secondarySort) {
```

Comment:
i think it's nice to keep the method generic here, it will make it easier to reuse in other services
And it makes me understand better the point of this method

Reply:
Difficult choice for me. I like to generalize but it seems unnecessarily complicated for a single column affected... For now I prefer to keep it like this.